### PR TITLE
Update the documentation and examples for the mismatch parameter site_name, add the version check in the main function.

### PR DESCRIPTION
**Pull request by Abhishek-121**
## Description

-- Update the documentation and examples for the mismatch parameter site_name.
-- Also add the version check in the main function for the sda fabric module support.
-- Remove the commented code.

## Type of Change
- [ yes] Bug fix
- [ no ] New feature
- [ no ] Breaking change
- [ yes ] Documentation update

## Checklist
- [ yes ] My code follows the style guidelines of this project
- [ yes ] I have performed a self-review of my own code
- [ yes ] I have commented my code, particularly in hard-to-understand areas
- [ yes ] I have made corresponding changes to the documentation
- [ yes ] My changes generate no new warnings
- [ no ] I have added tests that prove my fix is effective or that my feature works
- [ no ] New and existing unit tests pass locally with my changes
- [ no] Any dependent changes have been merged and published in downstream modules
- [ yes ] All the sanity checks have been completed and the sanity test cases have been executed

## Ansible Best Practices
- [ yes ] Tasks are idempotent (can be run multiple times without changing state)
- [ no ] Variables and secrets are handled securely (e.g., using `ansible-vault` or environment variables)
- [ yes ] Playbooks are modular and reusable
- [ no ] Handlers are used for actions that need to run on change

## Documentation
- [ yes ] All options and parameters are documented clearly.
- [ yes ] Examples are provided and tested.
- [ yes ] Notes and limitations are clearly stated.

### CODE REVIEW

File: plugins/modules/sda_fabric_virtual_networks_workflow_manager.py [hunk 1]
```diff
@@ -108,7 +108,7 @@
             type: list
             elements: dict
             suboptions:
-              site_name:
+              site_name_hierarchy:
```
Comment (madhansansel) [inline plugins/modules/sda_fabric_virtual_networks_workflow_manager.py:111]: Can't we say site_name itself like in the other module?

Comment (Abhishek-121) [reply plugins/modules/sda_fabric_virtual_networks_workflow_manager.py:111]: will decide on this.

File: plugins/modules/sda_fabric_sites_zones_workflow_manager.py [hunk 2]
```diff
@@ -754,43 +724,54 @@ def update_fabric_site(self, site, site_in_ccc):
...
+            payload = {"payload": update_site_params}
+            task_name = "update_fabric_site"
+            task_id = self.get_taskid_post_api_call("sda", task_name, payload)
 
-            if not response:
-                self.status = "failed"
-                self.msg = "Unable to fetch the task Id for the updation of fabric site as the 'update_fabric_site' response is empty."
-                self.log(self.msg, "ERROR")
+            if not task_id:
+                self.msg = "Unable to retrive the task_id for the task '{0}'.".format(task_name)
```
[the remark this answers is not in the document]
Comment (Abhishek-121) [reply plugins/modules/sda_fabric_sites_zones_workflow_manager.py]: updated.

File: plugins/modules/sda_fabric_sites_zones_workflow_manager.py [hunk 3]
```diff
@@ -754,43 +724,54 @@ def update_fabric_site(self, site, site_in_ccc):
...
-                    self.status = "success"
-                    self.update_site.append(site_name)
-                    self.log("Fabric site '{0}' updated successfully in the Cisco Catalyst Center".format(site_name), "INFO")
-                    break
-                time.sleep(1)
+            success_msg = "Fabric site '{0}' updated successfully in the Cisco Catalyst Center".format(site_name)
+            self.get_task_status_from_tasks_by_id(task_id, task_name, success_msg)
+            self.update_site.append(site_name)
+
+            # response = self.dnac._exec(
```
[the remark this answers is not in the document]
Comment (Abhishek-121) [reply plugins/modules/sda_fabric_sites_zones_workflow_manager.py]: removed.

File: plugins/modules/sda_fabric_sites_zones_workflow_manager.py [hunk 4]
```diff
@@ -820,54 +801,28 @@ def create_fabric_zone(self, zone):
...
-            )
-            self.log("Received API response from 'add_fabric_zone' for the site {0}: {1}".format(site_name, str(response)), "DEBUG")
-            response = response.get("response")
-
-            if not response:
-                self.status = "failed"
-                self.msg = "Unable to fetch the task Id for the creation of fabric zone as the 'add_fabric_zone' response is empty."
-                self.log(self.msg, "ERROR")
+            if not task_id:
+                self.msg = "Unable to retrive the task_id for the task '{0}'.".format(task_name)
```
[the remark this answers is not in the document]
Comment (Abhishek-121) [reply plugins/modules/sda_fabric_sites_zones_workflow_manager.py]: updated.

File: plugins/modules/sda_fabric_sites_zones_workflow_manager.py [hunk 5]
```diff
@@ -622,67 +622,37 @@ def create_fabric_site(self, site):
...
+            task_name = "add_fabric_site"
+            payload = {"payload": fabric_site_payload}
+            task_id = self.get_taskid_post_api_call("sda", task_name, payload)
 
-            if not response:
-                self.status = "failed"
-                self.msg = "No response received from 'add_fabric_site' API, task ID not retrieved."
-                self.log(self.msg, "ERROR")
+            if not task_id:
+                self.msg = "Unable to retrive the task_id for the task '{0}'.".format(task_name)
```
[the remark this answers is not in the document]
Comment (Abhishek-121) [reply plugins/modules/sda_fabric_sites_zones_workflow_manager.py]: updated.

**Review by madhansansel — APPROVED** [review]

**Pull request merged**